slackbridge/bots.py

```diff
@@ -129,17 +131,48 @@ def log(self, method, message):
         return method(full_message)
 
     def signedOn(self):
-        # If already registered, auth in
-        self.msg('NickServ', 'IDENTIFY {}'.format(self.nickserv_password))
-        # And if not, register for the first time
-        self.msg('NickServ', 'GROUP {} {}'.format(self.target_group_nick,
-                                                  self.nickserv_password))
+        self.nickserv_auth()
+
         for channel_name in self.joined_channels:
             self.log(log.msg, 'Joining #{}'.format(channel_name))
             self.join(channel_name)
 
         self.away('Default away for startup.')
 
+    def nickserv_auth(self):
+        # If already registered, authenticate yourself to Nickserv
+        self.msg('NickServ', 'IDENTIFY {}'.format(self.nickserv_password))
+        # And if not, register for the first time,
+        # if the nickname is the intended one.
+        if self.nickname == self.intended_nickname:
+            self.msg('NickServ', 'GROUP {} {}'.format(
+                self.target_group_nick, self.nickserv_password))
+
+    # Called whenever the client wants to set a nickname, such as on startup
+    # or if there's a collision.
+    def setNick(self, nickname):
+        super().setNick(nickname)
+        self.nickserv_auth()
+
+    def nickChanged(self, nick):
+        super().nickChanged(nick)
+        if nick != self.intended_nickname:
+            self.log(log.msg,
+                     'Attempting to change nick to {} in 10 seconds.'.format(
+                         self.intended_nickname))
+            reactor.callLater(10, self.setNick, self.intended_nickname)
+
+    # Called when an attempted nickname change collides with an existing one.
+    def irc_ERR_NICKNAMEINUSE(self, prefix, params):
+        # Appends an underscore to the intended nickname,
+        # and changes the bot's nickname to that
+        super().irc_ERR_NICKNAMEINUSE(prefix, params)
```

Comment:
I think this'll call `setNick` again [from the looks of the source](https://github.com/twisted/twisted/blob/b45bebbdf13f6d8639786d1023aa553ed75b5fc7/src/twisted/words/protocols/irc.py#L1946), which will then try to authenticate again with nickserv, so I think you'd want to change the nickserv authentication to only happen if it is indeed the correct nickname, otherwise it'll authenticate as an incorrect one instead (although it might later change).